pkg/executor/join/left_outer_semi_join_probe_test.go

```diff
@@ -31,8 +31,8 @@ import (
 
 // generate left outer semi join result using nested loop
 func genLeftOuterSemiJoinResult(t *testing.T, sessCtx sessionctx.Context, leftFilter expression.CNFExprs, leftChunks []*chunk.Chunk, rightChunks []*chunk.Chunk, leftKeyIndex []int, rightKeyIndex []int,
-	leftTypes []*types.FieldType, rightTypes []*types.FieldType, leftKeyTypes []*types.FieldType, rightKeyTypes []*types.FieldType, leftUsedColumns []int, rightUsedColumns []int,
-	otherConditions expression.CNFExprs, resultTypes []*types.FieldType) []*chunk.Chunk {
+	leftTypes []*types.FieldType, rightTypes []*types.FieldType, leftKeyTypes []*types.FieldType, rightKeyTypes []*types.FieldType, leftUsedColumns []int, otherConditions expression.CNFExprs,
+	resultTypes []*types.FieldType, isLeftOuter bool) []*chunk.Chunk {
```

Comment:
> Since it is only used in tests, we don't need to care too much about the efficiency. I think you can first generate the leftouter semi join results, then generate semi join results by filter the`match` column. In this way, you don't need to modify this function, and wil make the code more clear.

I think I can wrap this function in two function.